analysis/src/References.ml

```diff
@@ -468,15 +468,34 @@ let forLocalStamp ~full:{file; extra; package} stamp tip =
 let allReferencesForLocItem ~full:({file; package} as full) locItem =
   match locItem.locType with
   | TopLevelModule moduleName ->
-    let locs =
-      match Hashtbl.find_opt full.extra.fileReferences moduleName with
+    let otherModulesReferences =
+      package.localModules
+      |> Utils.filterMap (fun name ->
+             match ProcessCmt.fileForModule ~package name with
+             | None -> None
+             | Some file -> ProcessCmt.getFullFromCmt ~uri:file.uri)
+      |> List.map (fun full ->
+             match Hashtbl.find_opt full.extra.fileReferences moduleName with
+             | None -> []
+             | Some locs ->
+               locs
+               |> List.map (fun loc ->
+                      (Uri2.fromPath loc.Location.loc_start.pos_fname, [loc])))
+      |> List.flatten
+    in
+    let targetModuleReferences =
+      match Hashtbl.find_opt package.pathsForModule moduleName with
       | None -> []
-      | Some locs ->
-        locs
-        |> List.map (fun loc ->
-               (Uri2.fromPath loc.Location.loc_start.pos_fname, [loc]))
+      | Some paths -> (
+        let moduleSrcToRef src = (Uri2.fromPath src, [Utils.topLoc src]) in
+        match paths with
+        | Impl (_, None) -> []
```

Comment:
How about move this extraction of source files to `SharedTypes` alongside `getSrc`.